roles/power_aix_bootstrap/tasks/dnf_install.yml

```diff
@@ -0,0 +1,166 @@
+# Copyright (c) IBM Corporation 2021
+---
+
+#   Verify the AIX level is 7.3 or above. Bootstrap dnf is only supported in AIX 7.3 and up.
+- name: Verify the host is at least aix 7.3, if not, exit host
+  raw: "/usr/bin/oslevel  | /usr/bin/awk -F '.' ' { print $1 $2 }'"
+  register: output
+  changed_when: false
+
+- name: Fail if AIX version < 7.3
+  fail:
+     msg: "End playbook for {{ aix_host }}. Bootstrap dnf for AIX version \
+           {{ output.stdout_lines[0] }} is not supported."
+  when: (output.stdout_lines[0] | int) < 73
+
+#   CHECK for DNF on inventory host
+- block:
+
+     - name: Check for existence of dnf
+       raw: "rpm -q --quiet dnf && echo true || echo false"
+       register: dnf_exists
+       changed_when: false
+
+     - name: Print if dnf is installed
+       debug:
+          msg: "Exiting the playbook. dnf is already installed"
+       when: dnf_exists.stdout is search("true")
+
+     - name: Check for existence of yum
+       raw: "rpm -q --quiet yum && echo true || echo false"
+       register: yum_exists
+       changed_when: false
+       when: dnf_exists.stdout is search("false")
+
+  when: (output.stdout_lines[0] | int) >= 73
+
+# { Excute this block if dnf is not installed and the AIX version is igual or greater than 7.3
+- block:
+
+     # DOWNLOAD (local) install images
+     - name: Check for rpm_src file on localhost
+       stat:
+          path: "{{ download_dir }}/{{ rpm_src }}"
+       register: rpm_dest
+       delegate_to: localhost
+
+     - name: Download latest rpm (installp) fileset to the controller.
+       get_url:
+          url: "http://public.dhe.ibm.com/aix/freeSoftware/aixtoolbox/INSTALLP/ppc/{{ rpm_src }}"
+          dest: "{{ download_dir }}/{{ rpm_src }}"
+          mode: '0644'
+       delegate_to: localhost
+
+     - name: Check for "{{ dnf_src_71_72 }}" file on localhost
+       stat:
+          path: "{{ download_dir }}/{{ dnf_src_71_72 }}"
+       register: dnf_71_72_dest
+       delegate_to: localhost
+
+     - name: Check for "{{ dnf_src_73 }}" file on localhost
+       stat:
+          path: "{{ download_dir }}/{{ dnf_src_73 }}"
+       register: dnf_73_dest
+       delegate_to: localhost
+
+     - name: Download dnf bundle file for 7.1 and 7.2 to the controller
+       get_url:
+          url: "http://public.dhe.ibm.com/aix/freeSoftware/aixtoolbox/ezinstall/ppc/{{ dnf_src_71_72 }}"
+          dest: "{{ download_dir }}/{{ dnf_src_71_72 }}"
+          mode: '0644'
+       when: not dnf_71_72_dest.stat.exists
+       delegate_to: localhost
```

Comment:
We exit if aix version is 7.3 to install dnf. Then do we really need to download dnf for AIX 7.1, 7.2?

Reply:
@pvtorres is this comment resolved?